kubeflow/kubebench/kubebench-job.libsonnet

```diff
@@ -0,0 +1,355 @@
+                [
+                  buildStep(
+                    "wait-for-main-job",
+                    "main-job-monitor",
+                    argParams=[
+                      {
+                        name: "kf-job-manifest",
+                        value: "{{steps.run-configurator.outputs.parameters.kf-job-manifest}}",
+                      },
+                    ],
+                  ),
+                ],
+                [
+                  buildStep(
+                    "run-post-job",
+                    "post-job",
+                    argParams=[
+                      {
+                        name: "kf-job-manifest",
+                        value: "{{steps.run-configurator.outputs.parameters.kf-job-manifest}}",
+                      },
+                      {
+                        name: "experiment-id",
+                        value: "{{steps.run-configurator.outputs.parameters.experiment-id}}",
+                      },
+                    ],
+                  ),
+                ],
+                [
+                  buildStep(
+                    "run-reporter",
+                    "reporter",
+                    argParams=[
+                      {
+                        name: "kf-job-manifest",
+                        value: "{{steps.run-configurator.outputs.parameters.kf-job-manifest}}",
+                      },
+                      {
+                        name: "experiment-id",
+                        value: "{{steps.run-configurator.outputs.parameters.experiment-id}}",
+                      },
+                    ],
+                  ),
+                ],
+              ],
+            },
+            buildTemplate(
+              "configurator",
+              controllerImage,
+              [
+                "configurator",
+                "--template-ref=" + std.toString(mainJobKsPrototypeRef),
+                "--config=" + mainJobConfig,
+                "--namespace=" + namespace,
+                "--owner-references=" + std.toString(std.prune(ownerReferences)),
+                "--volumes=" + std.toString(std.prune(baseVols)),
+                "--volume-mounts=" + std.toString(std.prune(baseVolMnts)),
+                "--env-vars=" + std.toString(std.prune(baseEnvVars + expEnvVars(isConfigurator=true))),
+                "--manifest-output=" + manifestOutput,
+                "--experiment-id-output=" + experimentIdOutput,
+              ],
+              envVars=secretEnvVars + baseEnvVars,
+              volMnts=secretVolMnts + baseVolMnts,
+              outParams=[
+                {
+                  name: "kf-job-manifest",
+                  valueFrom: {
+                    path: manifestOutput,
+                  },
+                },
+                {
+                  name: "experiment-id",
+                  valueFrom: {
+                    path: experimentIdOutput,
+                  },
+                },
+              ],
+            ),
+            {
```

Comment:
What is this block for main-job? a template?

Reply:
this is a template. i did not use the buildTemplate() function here because it is a bit different from the rest of templates (generates a k8s resource vs. a job) hence the buildTemplate is not applicable here. i can wrap it in another buildResourceTemplate() function if that looks better.

Reply:
I see, should be fine.